flink-core/src/test/java/org/apache/flink/api/common/typeutils/TypeSerializerUpgradeTestBase.java

```diff
@@ -66,7 +58,22 @@
      * Creates a collection of {@link TestSpecification} which will be used as input for
      * parametrized tests.
      */
-    public abstract Collection<TestSpecification<?, ?>> createTestSpecifications() throws Exception;
+    public abstract Collection<TestSpecification<?, ?>> createTestSpecifications(
```

Comment:
Are we sure that we're not changing the behavior here? It looks like in the old implementations created specifications for each MIGRATION_VERSIONS entry. With the new implementation, we only do it for the version specified as a parameter in `MigrationTestsSnapshotGenerator`. :thinking: I might be wrong here. I just tried to reason it based on `CompositeTypeSerializerUpgradeTest`

Reply:
I think although Previously although the generating methods is executed for each specification, in each pass it executes the logic bound go `CURRENT_VERSION`: https://github.com/apache/flink/blob/057dbc5f62a5fce2b03de1c0a74a064ebbe9ac3e/flink-core/src/test/java/org/apache/flink/api/common/typeutils/TypeSerializerUpgradeTestBase.java#L271

Reply:
yikes, good catch. :-D that's a nasty one. The `getGenerate*FilePath` methods used `CURRENT_VERSION` as well instead of the version of the `testSpecification` parameter.